planner/core/exhaust_physical_plans.go

```diff
@@ -1820,17 +1820,20 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		return joins, true, nil
 	}
 
-	mergeJoins := p.GetMergeJoin(prop, p.schema, p.Stats(), p.children[0].statsInfo(), p.children[1].statsInfo())
-	if (p.preferJoinType&preferMergeJoin) > 0 && len(mergeJoins) > 0 {
-		return mergeJoins, true, nil
-	}
-	joins = append(joins, mergeJoins...)
+	if !p.isNAAJ() {
```

Comment:
We missed that if it's NAAJ, we should re-transform it to the old plan or just disable pushing down to tiflash for now?

Reply:
make sense.